study-builder/studies/pancan/study-events.conf

```diff
@@ -669,15 +669,37 @@
         "type": "ACTIVITY_INSTANCE_CREATION",
         "activityCode": ${id.act.about_cancer},
         "createFromAnswer": true,
-        "sourceQuestionStableId": ${id.q.primary_cancer_list},
+        "sourceQuestionStableId": ${id.q.primary_cancer_list_self},
         "targetQuestionStableId": ${id.q.diagnosis_type}
       },
       # If there's already about-cancer instances, then don't create any more.
-      "preconditionExpr": """!user.studies["cmi-pancan"].forms["ABOUT_CANCER"].hasInstance()""",
+      "preconditionExpr": """!user.studies["cmi-pancan"].forms["ABOUT_CANCER"].hasInstance()
+      && (user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("DIAGNOSED"))""",
       "maxOccurrencesPerUser": 1,
       "dispatchToHousekeeping": false,
       "order": 1
     },
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": ${id.act.release},
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "ACTIVITY_INSTANCE_CREATION",
+        "activityCode": ${id.act.about_cancer},
+        "createFromAnswer": true,
+        "sourceQuestionStableId": ${id.q.primary_cancer_list_child},
+        "targetQuestionStableId": ${id.q.diagnosis_type}
+      },
+      # If there's already about-cancer instances, then don't create any more.
+      "preconditionExpr": """!user.studies["cmi-pancan"].forms["ABOUT_CANCER"].hasInstance()
+      && (!user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("DIAGNOSED")
+            && user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("CHILD_DIAGNOSED"))""",
```

Comment:
I think we can reuse our `${_pex...}` strings here.

Reply:
> I am a bit worried about how some of the cancer option labels/names are super long..

@yufengwng it is generated automatically from the cancer titles. I can suggest that it can be shorten somehow. 
I can suggest 2 options:
- manually assign names for each cancer (it can take some time);
- during generation of the names limit with some max length (for example 25 or 30 chars). But here needs to think about the logic a bit: to avoid duplicates etc.

Reply:
Limiting by max length might not give the correct result. What if it cut off some of the abbreviations / aliases? Or if a single cancer name is longer than length and how it's split in half?